plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -135,6 +184,24 @@ func (c *podSecurityPolicyPlugin) Admit(a admission.Attributes) error {
 		return nil
 	}
 
+	// compute the context. If the current security context is valid, this call won't change it.
+	allowedPod, _, validationErrs, err := c.computeSecurityContext(a, pod, false)
```

Comment:
This is a pretty heavy operation (lots of deep copying...), and it would be nice if we could avoid doing it twice. Maybe we could keep a cache from mutation, and if none of the significant fields are updated skip this step? Might be a premature optimization, and definitely save for a separate PR...

Reply:
Was discussing with @deads2k to have some kind of information flow from the mutation phase to the validation phase. Something like an admission context. This would allow what you suggest easily.